parser-typechecker/src/Unison/Codebase/Path.hs

```diff
@@ -256,12 +258,24 @@ fromName' n = case take 1 (Name.toString n) of
     path = fromName n
     seq = toSeq path
 
-toName :: Path -> Name
-toName = Name.unsafeFromText . toText
+unsafeToName :: Path -> Name
```

Comment:
This function was partial but didn't indicate as such and caused me a few crashes while developing this, I marked it as partial and added a new safe `Maybe` version.
I also went through and patched some versions to the safe version wherever it was easy to do so.